### PR TITLE
EZP-21821: As an editor, I want to be able to edit content with an User field

### PR DESCRIPTION
**Pull request by dpobel**
JIRA: https://jira.ez.no/browse/EZP-21821

# Description

This patch provides the User edit view which allows to have complete form to edit user. It implements all the needed browser validation (including the login availability check added in https://github.com/ezsystems/ezpublish-kernel/pull/1415 / https://github.com/ezsystems/ez-js-rest-client/pull/65).

But actually, this PR is not be enough to be able to create/edit users because any content with a ezuser has to be saved through the User REST API.

Screencast: http://youtu.be/zodmsr0SP48

## Tasks

* [x] Basic user edit view
* [x] Check login availability
  * [x] Add a method for that in the JS REST client https://github.com/ezsystems/ez-js-rest-client/pull/65

# Tests

manual + unit tests

### CODE REVIEW

**Comment by dpobel** [comment]
ping @andrerom @mhyndle @yannickroger @StephaneDiot

File: Resources/public/js/views/fields/ez-user-editview.js [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+                this.set('emailErrorStatus', 'The email is not valid');
+            } else if ( emailRequired && !this._getInputFieldValue('email') ) {
+                this.set('emailErrorStatus', 'The email is required');
+            } else {
+                this.set('emailErrorStatus', false);
+            }
+        },
+
+        /**
+         * Validates the password inputs. Both value should be the same.
```
Comment (mhyndle) [inline Resources/public/js/views/fields/ez-user-editview.js]: nitpick: `values`

**Comment by mhyndle** [comment]
+1

**Comment by dpobel** [comment]
self note: I need to also apply the same change as in https://github.com/ezsystems/PlatformUIBundle/pull/298 to handle the field definition description in the new way.

**Comment by yannickroger** [comment]
+1

**Comment by andrerom** [comment]
+1

*Future possible followup: yellow feedback when password is weak.*

**Pull request merged**